### PR TITLE
Flask: limit number of families returned

### PR DESCRIPTION
**Pull request by Madelinehazel**
Related to #128 

`curl "http://localhost:3000/api/families?order=family_codename&starts_with=1&max_rows=1"`
returns:

```
[
  {
    "created": "Wed, 18 Nov 2020 21:12:38 GMT", 
    "created_by": 1, 
    "family_codename": "1000", 
    "family_id": 1, 
    "participants": [
      {
        "affected": true, 
        "created": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "created_by": 1, 
        "month_of_birth": "Sat, 01 Jan 2000 00:00:00 GMT", 
        "notes": "Extra info about sample here", 
        "participant_codename": "AA001", 
        "participant_id": 1, 
        "participant_type": "Proband", 
        "sex": "Female", 
        "solved": null, 
        "updated": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "updated_by": 1
      }, 
      {
        "affected": false, 
        "created": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "created_by": 1, 
        "month_of_birth": "Thu, 01 Jan 1970 00:00:00 GMT", 
        "notes": "", 
        "participant_codename": "AA002", 
        "participant_id": 2, 
        "participant_type": "Parent", 
        "sex": "Male", 
        "solved": null, 
        "updated": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "updated_by": 1
      }, 
      {
        "affected": false, 
        "created": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "created_by": 1, 
        "month_of_birth": "Sun, 01 Feb 1970 00:00:00 GMT", 
        "notes": "", 
        "participant_codename": "AA003", 
        "participant_id": 3, 
        "participant_type": "Parent", 
        "sex": "Female", 
        "solved": null, 
        "updated": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "updated_by": 1
      }, 
      {
        "affected": false, 
        "created": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "created_by": 1, 
        "month_of_birth": "Sun, 01 Mar 1970 00:00:00 GMT", 
        "notes": "", 
        "participant_codename": "AA004", 
        "participant_id": 4, 
        "participant_type": "Parent", 
        "sex": "Female", 
        "solved": null, 
        "updated": "Wed, 18 Nov 2020 21:12:38 GMT", 
        "updated_by": 1
      }, 
      {
        "affected": null, 
        "created": "Wed, 18 Nov 2020 21:13:02 GMT", 
        "created_by": 1, 
        "month_of_birth": null, 
        "notes": null, 
        "participant_codename": "AA3", 
        "participant_id": 5, 
        "participant_type": "Parent", 
        "sex": "Female", 
        "solved": null, 
        "updated": "Wed, 18 Nov 2020 21:13:02 GMT", 
        "updated_by": 1
      }
    ], 
    "updated": "Wed, 18 Nov 2020 21:12:38 GMT", 
    "updated_by": 1
  }
]
```

`curl "http://localhost:3000/api/families?order=skdhf"`

returns

`Column name must be one of ['family_id', 'family_codename', 'created', 'created_by', 'updated', 'updated_by']`, 400

`curl "http://localhost:3000/api/families?max_rows=a"`

returns

`Max rows must be a valid  integer`, 400

### CODE REVIEW

File: flask/app/families.py [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+    starts_with = f"{starts_with}%"
+    max_rows = request.args.get("max_rows", default=10000)
+    try:
+        int(max_rows)
+    except:
+        return "Max rows must be a valid  integer", 400
+
+    order = request.args.get("order")
+    if order:
+        columns = models.Family.__table__.columns._data.keys()
```
Comment (kevinlul) [inline flask/app/families.py:24]: I'm finding that just `.keys()` works, no need for `._data`

File: flask/app/families.py [hunk 2]
```diff
@@ -1,19 +1,43 @@
 from dataclasses import asdict
+import inspect
 
 from flask import abort, jsonify, request, Response, current_app as app
 from flask_login import login_user, logout_user, current_user, login_required
 from app import db, login, models
-from sqlalchemy.orm import joinedload
+from sqlalchemy.orm import joinedload, class_mapper
```
Comment (kevinlul) [inline flask/app/families.py:7]: Are these used?

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: flask/app/families.py (hunk 1 above)
Comment (Madelinehazel) [reply flask/app/families.py]: true!

File: flask/app/families.py (hunk 2 above)
Comment (Madelinehazel) [reply flask/app/families.py]: nope, removed

**Review by kevinlul — APPROVED** [review]
Seems okay, will wait on #184 due to conflicts

File: flask/app/families.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 from dataclasses import asdict
+import inspect
```
Comment (kevinlul) [inline flask/app/families.py:2]: unused

File: flask/app/families.py [hunk 4]
```diff
@@ -10,6 +11,21 @@
...
 @login_required
 def list_families():
+    starts_with = request.args.get("starts_with", default="", type=str)
+    starts_with = f"{starts_with}%"
+    max_rows = request.args.get("max_rows", default=100)
+    order_by_col = request.args.get("order", default="family_id", type=str)
+    try:
+        int(max_rows)
+    except:
+        return "Max rows must be a valid  integer", 400
```
Comment (kevinlul) [inline flask/app/families.py:21]: extra space

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: flask/app/families.py (hunk 3 above)
Comment (Madelinehazel) [reply flask/app/families.py]: resolved in [ae207ab](https://github.com/ccmbioinfo/ST2020/pull/195/commits/ae207abacd76978cf9409db8654f61ca59f5a1c5)

File: flask/app/families.py (hunk 4 above)
Comment (Madelinehazel) [reply flask/app/families.py]: resolved in [ae207ab](https://github.com/ccmbioinfo/ST2020/pull/195/commits/ae207abacd76978cf9409db8654f61ca59f5a1c5)

**Review by kevinlul — APPROVED** [review]

**Pull request merged**